server/src/controllers/Cart/UpdateCartController.js

```diff
@@ -0,0 +1,77 @@
+const { StatusCodes } = require('http-status-codes');
+
+class UpdateCartController {
+    constructor(redisClientService) {
+        this.redisClientService = redisClientService;
+    }
+
+    async index(req, res) {
+        const { cartId } = req.session;
+        const { id: productId } = req.params;
+        let { quantity, incrementBy } = req.body;
+
+        quantity = parseInt(quantity);
+        incrementBy = parseInt(incrementBy);
+
+        let productInStore = await this.redisClientService.jsonGet(`product:${productId}`);
+
+        if (!productInStore) {
+            return res.status(StatusCodes.BAD_REQUEST).send("Product with this id doesn't exist");
+        }
+
+        if (quantity === 0) {
```

Comment:
`if (!quantity)`
or maybe if it will be NaN then you do not want to call this `if` then it is ok

Reply:
I don't understand this comment, could you explain it differently?

Reply:
instead of `if (quantity === 0)` you can do `if (!quantity)`
but quantity my be `NaN` because you are parsing it to int few lines before (or maybe Im wrong
so i do not know if it make any difference if quantity is 0 or NaN because with `if (!quantity)` both return `true` and with `if (quantity === 0)` , value `0` returns `true` but `NaN` returns `false`

Reply:
done